catalog/CatalogRelation.cpp

```diff
@@ -93,9 +93,9 @@ CatalogRelation::CatalogRelation(const serialization::CatalogRelation &proto)
     }
   }
 
-  // Deserializing the indices of the relation
-  for (int i = 0; i < proto.index_size(); i++) {
-    indices_.push_back(proto.index(i));
+  // Deserializing the index scheme defined for the relation, if any
+  if (proto.has_index_scheme()) {
+    setIndexScheme(IndexScheme::ReconstructFromProto(proto.index_scheme()));
```

Comment:
I suggest to remove the method `setIndexScheme`, as it seems not to use by others.